x/concentrated-liquidity/position.go

```diff
@@ -126,18 +126,27 @@ func (k Keeper) GetAllPositionIdsForPoolId(ctx sdk.Context, prefix []byte, poolI
 	for ; iterator.Valid(); iterator.Next() {
 		key := iterator.Key()
 
-		// Extract the components from the key
-		parts := bytes.Split(key, []byte(types.KeySeparator))
-		if len(parts) != 4 {
-			return nil, fmt.Errorf("invalid key format: %s", key)
```

Comment:
How come this error wasn't popping? If the addr contained the key sep bytes shouldn't this have been 5 parts?

Reply:
There weren't tests reproducing this. If there were, it would. I will add a new test case prior to merge